WooCommerce/Classes/ServiceLocator/ProductImageUploader.swift

```diff
@@ -80,9 +83,17 @@ final class ProductImageUploader: ProductImageUploaderProtocol {
         guard let handler = actionHandlersByProduct[key] else {
             return
         }
+
+        // Update the product ID of handler to make sure that future product image uploads use the `remoteProductID` instead of `localProductID`
+        handler.updateProductID(remoteProductID)
+
         actionHandlersByProduct.removeValue(forKey: key)
         let keyWithRemoteProductID = ProductKey(siteID: siteID, productID: remoteProductID, isLocalID: false)
         actionHandlersByProduct[keyWithRemoteProductID] = handler
+
+        updateProductIDOfImagesUploadedUsingLocalProductID(siteID: siteID,
```

Comment:
❓ since there could be images pending upload at the time of this call, those images in progress might still be missing a parent ID (as mentioned in the test result in the review summary). Does it make sense to call this in `saveProductImagesWhenNoneIsPendingUploadAnymore` when there are no images pending upload?